### DISCUSSION TITLE
Accessing custom values in registries

### DISCUSSION THREAD

**Discussion by BitCrack** [Minestom/Minestom#854] [Help] [answered]
I'm generating additional data myself in my own MinestomDataGenerator fork (I haven't tried PR'ing any of the changes yet because for now I'm still experimenting).
I was able to access this data like so
```java
//...
var registry = enchantment.registry();
var categoryId = registry.getString("category");
```
After https://github.com/Minestom/Minestom/pull/651 the solution above is no longer possible as all getter methods have been removed from `Entry`.
There is `Entry#custom`, but after looking at all subclasses of `Entry`, they just pass null and there's no way to change it because it's final without any setters. Even then I don't believe it'd make sense (and I don't think that was its purpose) for data coming from MinestomDataGenerator to be inside of that.
I would assume that the best "fix" (at least for now) is to also make a fork for Minestom?

**Comment by TheMode** [comment, marked as answer]
For now you will indeed need a fork.

The `custom` registry will be available for expansion once third party (extensions etc) get the ability to introduce new fields.
